libs/wingsdk/src/shared/bundling.ts

```diff
@@ -47,17 +57,41 @@ export function createBundle(entrypoint: string, outputDir?: string): Bundle {
   // the bundled contains line comments with file paths, which are not useful for us, especially
   // since they may contain system-specific paths. sadly, esbuild doesn't have a way to disable
   // this, so we simply filter those out from the bundle.
-  const outlines = readFileSync(outfile, "utf-8").split("\n");
-  const isNotLineComment = (line: string) => !line.startsWith("//");
-  const final = outlines.filter(isNotLineComment).join("\n");
-  writeFileSync(outfile, final);
+  const output = esbuild.outputFiles[1];
+  let fileContents = new TextDecoder().decode(output.contents);
+
+  // sourcemap hacks for winglang:
+  // source paths in sourcemap are incorrect and need to be fixed
+  const sourcemapData = JSON.parse(
+    new TextDecoder().decode(esbuild.outputFiles[0].contents)
+  );
+  // unrandomize the sourceRoot
+  sourcemapData.sourceRoot = normalPath(sourcemapData.sourceRoot).replace(
+    /\.\d+\.tmp\/\.wing\//g,
+    "/.wing/"
+  );
```

Comment:
ℹ️ Along with some other changes in this file, this probably fixes #3647. I didn't added anything to verify that though